src/Compilers/CSharp/Portable/Emitter/Model/PEModuleBuilder.cs

```diff
@@ -1009,7 +1006,7 @@ internal static Cci.IGenericParameterReference Translate(TypeParameterSymbol par
             bool needDeclaration = false)
         {
             Debug.Assert(fieldSymbol.IsDefinitionOrDistinct());
-            Debug.Assert(!fieldSymbol.IsVirtualTupleField && (object)(fieldSymbol.TupleUnderlyingField ?? fieldSymbol) == fieldSymbol, "tuple fields should be rewritten to underlying by now");
```

Comment:
> && (object)(fieldSymbol.TupleUnderlyingField ?? fieldSymbol) == fieldSymbol [](start = 57, length = 76)

I think we should keep this condition, it is still bad to violate it.

Reply:
In fact, I think we should make the assert even stronger and make sure we don't run into a ```TupleErrorFieldSymbol```.

---
In reply to: [426203778](https://github.com/dotnet/roslyn/pull/44231#discussion_r426203778) [](ancestors = 426203778)

Reply:
It doesn't look like the thread is actually addressed completely and no other response provided.

---
In reply to: [426264817](https://github.com/dotnet/roslyn/pull/44231#discussion_r426264817) [](ancestors = 426264817,426203778)